Yosemite/Yosemite/Actions/InboxNotesAction.swift

```diff
@@ -22,6 +22,12 @@ public enum InboxNotesAction: Action {
                           noteID: Int64,
                           completion: (Result<Bool, Error>) -> ())
 
+    /// Dismiss all `InboxNote`s.
+    /// This marks all notifications is_deleted field to true and the inbox notes will be deleted locally.
+    ///
+    case dismissAllInboxNotes(siteID: Int64,
+                              completion: (Result<Bool, Error>) -> ())
```

Comment:
nit: since the success case doesn't return a boolean value that is `false` in the store implementation, does it make sense to have the result success type as `Void`?

```suggestion
                              completion: (Result<Void, Error>) -> ())
```